internal/batch/batch_manager.go

```diff
@@ -588,3 +594,30 @@ func (bm *batchManager) LoadContexts(ctx context.Context, payload *DispatchPaylo
 	}
 	return nil
 }
+
+func (bm *batchManager) CancelBatch(ctx context.Context, batchID string) error {
+	id, err := fftypes.ParseUUID(ctx, batchID)
+	if err != nil {
+		return err
+	}
+	bp, err := bm.database.GetBatchByID(ctx, bm.namespace, id)
+	if err != nil {
+		return err
+	}
+	batch, err := bm.data.HydrateBatch(ctx, bp)
+	if err != nil {
+		return err
+	}
+	if len(batch.Payload.Messages) == 0 {
+		return i18n.NewError(ctx, coremsgs.MsgErrorLoadingBatch)
+	}
+	msg := batch.Payload.Messages[0]
+	processor, err := bm.getProcessor(msg.Header.TxType, msg.Header.Type, msg.Header.Group, msg.Header.SignerRef.Author, false)
```

Comment:
Just checking - what is it that means we're sure that a processor will be active in this case (so we don't need to create one)?

Reply:
... ah ok, I can see that `cancelFlush` is only going to do anything meaningful if it's busy trying to flush. So the `processor == nil` case means there's nothing to do.

Reply:
We can only cancel a batch that is stuck in a processor. So if the relevant processor is not active, we simply return an error.